### ISSUE TITLE
aws_sns_topic_subscription diffs didn't match during apply.

### ISSUE THREAD

**Issue by dansteen** [hashicorp/terraform#5677]
Got the following error when running an apply (this was for an update)

```
Error applying plan:

1 error(s) occurred:

* aws_sns_topic_subscription.assessment-trait-scores-scaler: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.13
    Resource ID: aws_sns_topic_subscription.assessment-trait-scores-scaler
    Mismatch reason: attribute mismatch: endpoint
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"endpoint":*terraform.ResourceAttrDiff{Old:"arn:aws:lambda:us-east-1:xxxxxxx:function:assessment-trait-scores-scaler-prod", New:"${aws_lambda_function.assessment-trait-scores-scaler.arn}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.
```

The config for this looks like this:

```
resource "aws_sns_topic_subscription" "assessment-trait-scores-scaler" {
  topic_arn = "${aws_sns_topic.assessment-trait-scores_alarm.arn}"
  protocol  = "lambda"
  endpoint_auto_confirms = false
  endpoint  = "${aws_lambda_function.assessment-trait-scores-scaler.arn}"
}
```

When I run "terraform apply" a second time everything works fine.  I see this whenever I update the aws_lambda_function being referenced as an endpoint.

v0.6.13

Thanks!

**Comment by gonzaloserrano** [comment]
More or less same thing after upgrading from 0.6.13 to .14:

```
* aws_sns_topic_subscription.push-notifier-lambda-sns-delivery-failures: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.14
    Resource ID: aws_sns_topic_subscription.push-notifier-lambda-sns-delivery-failures
    Mismatch reason: attribute mismatch: endpoint
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"endpoint":*terraform.ResourceAttrDiff{Old:"arn:aws:lambda:us-east-1:646470220325:function:snsDeliveryFailures", New:"${aws_lambda_function.push-notifier-sns-delivery-failures.arn}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}
```

Regards,

**Comment by pporada-gl** [comment]
I'm running into this on v0.6.16
```
1 error(s) occurred:

* aws_sns_topic_subscription.lambda: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.16
    Resource ID: aws_sns_topic_subscription.lambda
    Mismatch reason: attribute mismatch: endpoint
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"endpoint":*terraform.ResourceAttrDiff{Old:"arn:aws:lambda:us-east-1:REDACTEDREDACTED:function:ASG-Slack-Notify", New:"${aws_lambda_function.slack_notify.arn}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}
```

**Comment by mitchellh** [comment]
Hello! This should be fixed in later versions of Terraform (0.7.11 at time of writing). 

Its hard to know 100% for a couple reasons:

  * This is on a now much-older version of Terraform. This is our fault, not yours, for not trying sooner, but we've fixed many diff mismatch issues since then.

  * We've fixed issues with exact error messages as this in those versions. 

Please try again! If the issue persists please just open a new issue. Thanks!
